### ISSUE TITLE
Let CSV importer support tab-separated files

### ISSUE THREAD

**Issue by l9i** [beancount/beangulp#85]
Would it be possible for the CSV importer to support tab-separated files as well? Currently the 'text/csv' MIME type [seems hardcoded](https://github.com/beancount/beancount/blob/f02f530d18fd832752e0b51683bb7c4ac832feeb/beancount/ingest/importers/csv.py#L177).

While I could monkey patch it with `self.remap['mime'] = [re.compile('text/tab-separated-values')]` in my subclass' `__init__()`, it'd be rather ugly.

**Comment by l9i** [comment]
@jarifuri, I don't think this problem can be solved by passing `csv_dialect` since the MIME type matcher always expects `text/csv`:  https://github.com/beancount/beangulp/blob/44d235b80cec8d98ed2ba4dfc368b741f3c77abd/beangulp/importers/csv.py#L181

**Comment by dnicolodi** [comment]
I don't see anything wrong with overriding the mime type matching in a subclass. Anyhow, the this CSV importer is deprecated and replaced by `beangulp.importers.csvbase.Importer`.
